ActionSheetProvider.js

```diff
@@ -2,27 +2,31 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import ActionSheet from './ActionSheet';
 
+import { Provider } from './Context';
+
 export default class ActionSheetProvider extends React.Component {
   static propTypes = {
     children: PropTypes.any,
   };
 
-  static childContextTypes = {
-    showActionSheetWithOptions: PropTypes.func,
-  };
+  constructor(props) {
+    super(props);
+
+    this._actionSheetRef = React.createRef();
+    this.showActionSheetWithOptions = this.showActionSheetWithOptions.bind(this);
+  }
 
-  getChildContext() {
-    return {
-      showActionSheetWithOptions: (...args) =>
-        this._actionSheetRef.showActionSheetWithOptions(...args),
-    };
+  showActionSheetWithOptions(...args) {
```

Comment:
Can you change this function to arrow function so you don't have to do this `this.showActionSheetWithOptions = this.showActionSheetWithOptions.bind(this);`

Reply:
ah, use bind allow another class can be extended and reuse this method.

https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1

Reply:
We are not calling this method in loop or anything so performance isn't the issue :D